dask/threaded.py

```diff
@@ -5,21 +5,28 @@
 """
 from __future__ import absolute_import, division, print_function
 
+from collections import defaultdict
 from multiprocessing.pool import ThreadPool
+import threading
 from threading import current_thread
+
 from .async import get_async
 from .compatibility import Queue
 from .context import _globals
 from .utils_test import inc, add  # noqa: F401
 
 
-default_pool = ThreadPool()
-
-
 def _thread_get_id():
     return current_thread().ident
 
 
+default_pool = ThreadPool()
+default_thread = _thread_get_id()
+main_thread = current_thread()
+
+pools = defaultdict(dict)
```

Comment:
During cleanup it is convenient to have all threadpools indexed by thread.  Could flatten though.  In the common case this isn't large.